web/themes/contrib/civictheme/includes/form_element.inc

```diff
@@ -0,0 +1,425 @@
+<?php
+
+/**
+ * @file
+ * Form elements.
+ *
+ * Drupal uses 'form element' as a unit of rendering for form elements: it is
+ * a wrapper around specific form controls (input, select, textarea, etc.).
+ *
+ * For some form elements, like checkboxes and radios, the form element is
+ * rendered as a fieldset element first, and then the actual form controls are
+ * rendered inside the fieldset element.
+ *
+ * CivicTheme uses slightly different approach to render form elements: it uses
+ * a centralised template (factory) called "Field" which is responsible for
+ * rendering form elements in a consistent way - this allows to strictly align
+ * form elements with the design system.
+ *
+ * The code below implements the necessary preprocessors to redirect rendering
+ * to the dedicated form element template and allow to preprocess form elements
+ * based on their type.
+ *
+ * Note that sub-themes can opt-out from this behavior by implementing
+ * hook_theme_suggestions_form_element_alter() and
+ * hook_theme_suggestions_fieldset_alter() and removing the suggestions
+ * added here based on the required conditions.
+ *
+ * A render pipeline for form_element and fieldset looks like this:
+ *
+ * +-------------------------------------------------+
+ * | civictheme_theme_suggestions_form_element_alter |
+ * | civictheme_theme_suggestions_fieldset_alter     |
+ * |    - Check element type                         |
+ * |    - Add suggestions                            |
+ * +-------------------------------------------------+
+ *          |
+ *          v
+ * +----------------------------------------------------+
+ * | Theme Hook Suggestions Generated                   |
+ * | - form_element__civictheme_field                   |
+ * | - form_element__civictheme_field__<type>           |
+ * | - fieldset__form_element__civictheme_field         |
+ * | - fieldset__form_element__civictheme_field__<type> |
+ * +----------------------------------------------------+
+ *          |
+ *          v
+ * +-----------------------------------------------------+
+ * | Sub-theme hook_theme_suggestions_form_element_alter |
+ * |    - Check opt-out condition                        |
```

Comment:
The opt out condition is delete the form suggestion?

Reply:
>  
```
 * Note that sub-themes can opt-out from this behavior by implementing
 * hook_theme_suggestions_form_element_alter() and
 * hook_theme_suggestions_fieldset_alter() and removing the suggestions
 * added here based on the required conditions.
```

Reply:
```php
function yourtheme_theme_suggestions_form_element_alter($suggestions){
  if ($variables['element']['#type'] == 'select'){
    unset($suggestions[array_search($suggestions, 'form_element__civictheme_field' )]);
  }
}
```